besu/src/main/java/org/hyperledger/besu/cli/subcommands/storage/TrieLogHelper.java

```diff
@@ -88,11 +95,23 @@ void prune(
         numberOfBatches,
         batchFileNameBase);
 
-    if (rootWorldStateStorage.streamTrieLogKeys(layersToRetain).count() == layersToRetain) {
-      deleteFiles(batchFileNameBase, numberOfBatches);
-      LOG.info("Prune ran successfully. Enjoy some disk space back! \uD83D\uDE80");
+    // Should only be layersToRetain left but loading extra just in case of an unforeseen bug
+    final long countAfterPrune =
+        rootWorldStateStorage
+            .streamTrieLogKeys(layersToRetain + DEFAULT_BONSAI_TRIE_LOG_PRUNING_WINDOW_SIZE)
+            .count();
+    if (countAfterPrune == layersToRetain) {
+      if (deleteFiles(batchFileNameBase, numberOfBatches)) {
+        LOG.info("Prune ran successfully. Enjoy some disk space back! \uD83D\uDE80");
+      } else {
+        throw new IllegalStateException(
+            "There was an error deleting the trie log backup files. Please ensure besu is working before deleting them manually.");
+      }
     } else {
-      LOG.error("Prune failed. Re-run the subcommand to load the trie logs from file.");
+      throw new IllegalStateException(
+          String.format(
+              "Remaining trie logs (%d) did not match %s (%d). Trie logs backup files have not been deleted, it is safe to rerun the subcommand.",
+              countAfterPrune, BONSAI_STORAGE_FORMAT_MAX_LAYERS_TO_LOAD, layersToRetain));
```

Comment:
Should we remove BONSAI_STORAGE_FORMAT_MAX_LAYERS_TO_LOAD from this exception message?

Reply:
why?

Should look like this 
> Remaining trie logs (0) did not match --bonsai-historical-block-limit (3). Trie logs backup files have not been deleted, it is safe to rerun the subcommand

Reply:
Nevermind misread that and thought we were using the default value of that option.